regulations/tasks.py

```diff
@@ -35,47 +37,54 @@ def submit_comment(self, body, files):
-                ('uploadedFile', ('comment.pdf', comment)),
-                ('general_comment', 'See attached comment.pdf'),
-            ]
-
-            # Add other submitted fields
-            fields.extend([
-                (name, value)
-                for name, value in six.iteritems(body)
-                if name != 'general_comment'
-            ])
-            fields.extend(attachments)
-
-            data = MultipartEncoder(fields)
-            response = requests.post(
-                settings.REGS_GOV_API_URL,
-                data=data,
-                headers={
-                    'Content-Type': data.content_type,
-                    'X-Api-Key': settings.REGS_GOV_API_KEY,
-                }
-            )
-            if response.status_code != requests.codes.created:
-                logger.warn("Post to regulations.gov failed: %s %s",
-                            response.status_code, response.text)
-                raise self.retry()
-            logger.info(response.text)
-            return response.json()
-    except (ClientError, RequestException) as exc:
-        logger.exception(exc)
-        raise self.retry(exc=exc)
+        html = json_to_html(body)
+        try:
+            with html_to_pdf(html) as comment, \
+                    build_attachments(files) as attachments:
+                fields = [
+                    ('comment_on', settings.COMMENT_DOCUMENT_ID),
+                    # TODO: Ensure this name is unique
+                    ('uploadedFile', ('comment.pdf', comment)),
+                    ('general_comment', 'See attached comment.pdf'),
+                ]
+
+                # Add other submitted fields
+                fields.extend([
+                    (name, value)
+                    for name, value in six.iteritems(body)
+                    if name != 'general_comment'
+                ])
+                fields.extend(attachments)
+
+                data = MultipartEncoder(fields)
+                response = requests.post(
+                    settings.REGS_GOV_API_URL,
+                    data=data,
+                    headers={
+                        'Content-Type': data.content_type,
+                        'X-Api-Key': settings.REGS_GOV_API_KEY,
+                    }
+                )
+                if response.status_code != requests.codes.created:
+                    logger.warn("Post to regulations.gov failed: %s %s",
+                                response.status_code, response.text)
+                    raise self.retry()
+                logger.info(response.text)
+                return response.json()
+        except (RequestException, ClientError):
+            logger.exception("Unable to submit comment")
+            raise self.retry()
+    except MaxRetriesExceededError:
+        message = "Exceeded retries, saving failed submission"
+        logger.error(message)
+        failed_submission = FailedCommentSubmission(body=body, files=files)
```

Comment:
Minor, but since `files` is derived from `body`, do we need to store both? Also minor: `body` and `files` are a dict and a list, respectively, but we're putting them into `TextField`s. Django seems to stringify them on save, although it doesn't parse them on load. I think it would be clearer to explicitly stringify in our code and not rely on Django to do this.